### PR TITLE
add api for private link resources

### PR DESCRIPTION
**Pull request by gossion**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2020-09-01/managedClusters.json [hunk 1]
```diff
@@ -2985,6 +3080,61 @@
           "description": "The private link service connection description."
         }
       }
+    },
+    "PrivateLinkResourcesListResult": {
```
Comment (gossion) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2020-09-01/managedClusters.json:3084]: for reviewers: we has only 1 item in the list currently, so pagination is not needed.

**Comment by gossion** [comment]
@NullMDR any idea why "automation - sdk" failed?

**Review by feiskyer — APPROVED** [review]

**Comment by NullMDR** [comment]
@gossion In .net sdk build log

```
[MSBuild]      3>Tests/Tests/ContainerServiceOpsTests.cs(28,76): error CS1061: 'ContainerServiceClient' does not contain a definition for 'ContainerServices' and no accessible extension method 'ContainerServices' accepting a first argument of type 'ContainerServiceClient' could be found (are you missing a using directive or an assembly reference?) [/z/work/azure-sdk-for-net/sdk/containerservice/Microsoft.Azure.Management.ContainerService/tests/Microsoft.Azure.Management.ContainerService.Tests.csproj]
```

It's related to test build failure which I assume should be fine?

**Comment by gossion** [comment]
thanks @NullMDR , can you review the PR and sign off if there is no issue.
btw, the PR has label 'WaitForARMFeedback', who from ARM team can take a look?

**Comment by NullMDR** [comment]
@gossion Oh for your PR you don't need ARM review but please fix spellcheck https://github.com/Azure/azure-rest-api-specs/pull/10548/checks?check_run_id=1010987618

**Comment by gossion** [comment]
thanks @NullMDR , did not notice that error.
```
Unknown word (azpodpolicy)
```
 it is not my change, but I just updated custom-words.txt to fix the error. :)

**Comment by gossion** [comment]
@NullMDR spellcheck  error fixed, can you help to check and merge the pr?

**Review by PhoenixHe-NV — APPROVED** [review]

**Pull request merged**